tests/components/FilterSearch.test.tsx

```diff
@@ -0,0 +1,299 @@
+import { FilterSearch } from '../../src/components/FilterSearch';
+import { FilterSearchResponse } from '@yext/answers-headless-react';
+import { render, screen } from '@testing-library/react';
+import * as useSynchronizedRequestFunctions from '../../src/hooks/useSynchronizedRequest';
+import userEvent from '@testing-library/user-event';
+import * as searchOperations from '../../src/utils/search-operations';
+import { mockAnswersActions } from '../__utils__/mocks';
+
+jest.mock('@yext/answers-headless-react');
+
+const mockedFilterSearchResponse: FilterSearchResponse = {
+  sections: [{
+    label: 'People',
+    results: [
+      { value: 'John Doe',
+        filter: { fieldId: null, matcher: null, value: 'test' } },
+      { value: 'Jane Doe',
+        filter: { fieldId: null, matcher: null, value: 'test' } }
+    ]
+  },
+  {
+    label: 'Condiments',
+    results: [
+      { value: 'Jam',
+        filter: { fieldId: null, matcher: null, value: 'test' } }
+    ]
+  }],
+  uuid: null
+};
+
+const mockedFilterSearchResponseNoLabels: FilterSearchResponse = {
+  sections: [{
+    results: [
+      { value: 'John Doe',
+        filter: { fieldId: null, matcher: null, value: 'test' } },
+      { value: 'Jane Doe',
+        filter: { fieldId: null, matcher: null, value: 'test' } }
+    ]
+  }],
+  uuid: null
+};
+
+const mockedFilterSearchResponseNoResults: FilterSearchResponse = {
+  sections: [{
+    results:[]
+  }],
+  uuid: null
+};
+
+const mockedExecuteFilterSearch = jest.fn();
+const setFilterOption = jest.fn();
+const setOffset = jest.fn();
+const searchFieldsProp = {
+  fieldApiName: 'name',
+  entityType: 'ce-person'
+};
+
+describe('tests using data with section labels', () => {
+  beforeEach(() => {
+    mockAnswersActions({
+      setFilterOption,
+      setOffset
+    });
+
+    jest.spyOn(searchOperations, 'executeSearch').mockImplementation();
+    jest.spyOn(useSynchronizedRequestFunctions, 'useSynchronizedRequest').mockImplementation(() => {
+      return [mockedFilterSearchResponse, mockedExecuteFilterSearch, null];
+    });
+  });
+
+  it('renders the filter search bar and the "Filter" label', () => {
+    render(<FilterSearch searchFields={[searchFieldsProp]}/>);
+
+    const label = 'Filter';
+    const labelElement = screen.getByText(label);
+    const searchBarElement = screen.getAllByRole('textbox');
+
+    expect(labelElement).toBeDefined();
+    expect(searchBarElement.length).toBe(1);
+  });
+
+  it('calls executeFilterSearch every time a charachter is entered', () => {
+    render(<FilterSearch searchFields={[searchFieldsProp]}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    expect(searchBarElement).toBeDefined();
```

Comment:
We probably don't need an `expect(searchBarElement).toBeDefined();` in tests other than `renders the filter search bar and the "Filter" label`. Ideally we don't have multiple tests that have expect() statements on the same "thing" though that's a bit nebulous, and I personallyy lean towards just spamming expect() statements since I'm more worried about missing test cases than being overly redundant, but it's something to keep in mind!

Reply:
Actually I see that below you started to just drop the expect searchBarElement to beDefined already!

Reply:
removed it from the unnecessary functions